src/widgets/spoolman/component.jsx

```diff
@@ -0,0 +1,41 @@
+import { useTranslation } from "next-i18next";
+
+import Container from "components/services/widget/container";
+import Block from "components/services/widget/block";
+import useWidgetAPI from "utils/proxy/use-widget-api";
+
+export default function Component({ service }) {
+  const { t } = useTranslation();
+  const { widget } = service;
+
+  const { data: spoolData, error: spoolError } = useWidgetAPI(widget, "spools");
+
+  if (spoolError) {
+    return <Container service={service} error={spoolError} />;
+  }
+
+  if (!spoolData) {
+    return (
+      <Container service={service}>
+        <Block label="spoolman.spool1" />
+        <Block label="spoolman.spool2" />
+      </Container>
+    );
+  }
+
+  if (spoolData.error || spoolData.message) {
+    return <Container service={service} error={spoolData?.error ?? spoolData} />;
+  }
+
```

Comment:
```suggestion
  if (spoolData.length === 0) {
    return (
      <Container service={service}>
        <Block label="spoolman.noSpools" />
      </Container>
    );
  }

  if (spoolData.length > 4) spoolData = spoolData.slice(0, 4);

```

1. I would handle the no spools case (need to add entry to common.json)
2. We dont allow more than 4 blocks, as noted in the guidelines. So up to you how to handle it, at minimum just slice the 4, or you can change the widget in that case (e.g. show a block with total, some other info, etc)